org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java

```diff
@@ -14,72 +14,22 @@
 
 import java.nio.file.Path;
 import java.util.List;
-import java.util.Objects;
 import java.util.stream.Stream;
 
 import org.eclipse.aether.graph.DependencyNode;
 
 import aQute.bnd.osgi.Jar;
 
-public final class WrappedBundle {
-
-	private DependencyNode node;
-	private List<WrappedBundle> depends;
-	private String instructionsKey;
-	private Path file;
-	private Jar jar;
-	private List<ProcessingMessage> messages;
-
-	WrappedBundle(DependencyNode node, List<WrappedBundle> depends, String key, Path file, Jar jar,
-			List<ProcessingMessage> messages) {
-		this.node = node;
-		this.depends = depends;
-		this.instructionsKey = key;
-		this.file = file;
-		this.jar = jar;
-		this.messages = messages;
-	}
-
-	String getInstructionsKey() {
-		return instructionsKey;
-	}
-
-	Jar getJar() {
-		return jar;
-	}
-
-	/**
-	 * @return the file where the wrappes bundle is located
-	 */
-	public Path getFile() {
-		return file;
-	}
+public record WrappedBundle(DependencyNode node, List<WrappedBundle> depends, String instructionsKey,
```

Comment:
This should not be a record as it is not a pure container, this makes all its members public to the outside world what is not desired.